modules/tinymce/src/plugins/advlist/test/ts/browser/ToolbarButtonStructureTest.ts

```diff
@@ -0,0 +1,71 @@
+import { UnitTest } from "@ephox/bedrock-client";
+import { Pipeline, Log, Chain, NamedChain, Waiter, Assertions, ApproxStructure } from '@ephox/agar';
+import AdvListPlugin from 'tinymce/plugins/advlist/Plugin';
+import ListsPlugin from 'tinymce/plugins/lists/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+import { Editor } from '@ephox/mcagar';
+import { Element } from '@ephox/sugar';
+
+UnitTest.asynctest('browser.tinymce.plugins.advlist.ToolbarButtonStructureTest', (success, failure) => {
+  AdvListPlugin();
+  ListsPlugin();
+  Theme();
+
+  const cGetContainer = Chain.mapper((editor: any) => Element.fromDom(editor.editorContainer));
+
+  Pipeline.async({}, [
+    Log.step('TBA', 'Test that one list type = toolbar button NOT splitbutton', Chain.asStep({}, [
+      Editor.cFromSettings({
+        theme: 'silver',
+        plugins: 'lists advlist',
+        toolbar: 'numlist',
+        advlist_number_styles: 'default',
+        menubar: false,
+        statusbar: false,
+        base_url: '/project/tinymce/js/tinymce'
+      }),
+      NamedChain.asChain([
+        NamedChain.direct(NamedChain.inputName(), Chain.identity, 'editor'),
+        NamedChain.direct('editor', cGetContainer, 'editorContainer'),
+        NamedChain.direct('editorContainer', Waiter.cTryUntil('', Assertions.cAssertStructure(
+          'Check p>strong element path',
+          ApproxStructure.build((s, str, arr) => s.element('div', {
```

Comment:
This is way more than what's needed, as you don't need to assert the entire editor structure. So you could simplify all this `NamedChain` code to be at least half the size:

```
    Log.chainsAsStep('TBA', 'Test that one list type = toolbar button NOT splitbutton', [
      Editor.cFromSettings({
        theme: 'silver',
        plugins: 'lists advlist',
        toolbar: 'numlist',
        advlist_number_styles: 'default',
        menubar: false,
        statusbar: false,
        base_url: '/project/tinymce/js/tinymce'
      }),
      NamedChain.asChain([
        NamedChain.writeValue('body', Body.body()),
        NamedChain.direct('body', UiFinder.cFindIn('.tox-editor-header .tox-toolbar .tox-toolbar__group'), 'toolbarGroup'),
        NamedChain.read('toolbarGroup', Waiter.cTryUntil('', Assertions.cAssertStructure(
          'Check lists toolbar button structure',
          ApproxStructure.build((s, str, arr) => s.element('div', {
            classes: [ arr.has('tox-toolbar__group') ],
            children: [
              s.element('button', {
                classes: [ arr.has('tox-tbtn') ],
                attrs: {
                  title: str.is('Numbered list')
                }
              })
            ]
          }))
        ))),
        NamedChain.outputInput
      ]),
      Editor.cRemove
    ])
```

Reply:
Also worth noting the editor wasn't being cleaned up in the previous code, so the above handles that.